src/pdb/mod.rs

```diff
@@ -358,63 +363,74 @@ pub struct IndexPageHeader {
     pub first_empty: u16,
 }
 
+impl IndexPageHeader {
+    /// Size of the index page header in bytes.
+    pub const BINARY_SIZE: u32 = 28;
+}
+
 /// The content of an index page.
-#[binread]
+#[binrw]
 #[derive(Debug, PartialEq, Eq, Clone)]
 #[br(little)]
+#[bw(little, import { page_size: u32 })]
 pub struct IndexPageContent {
     /// The header of the index page.
     pub header: IndexPageHeader,
 
     /// The index entries.
     #[br(count = header.num_entries)]
     pub entries: Vec<IndexEntry>,
-}
-
-impl BinWrite for IndexPageContent {
-    type Args<'a> = (u32,);
 
-    fn write_options<W: Write + Seek>(
-        &self,
-        writer: &mut W,
-        endian: Endian,
-        (page_size,): Self::Args<'_>,
-    ) -> BinResult<()> {
-        let page_content_start_pos = writer.stream_position()?;
-
-        self.header.write_options(writer, endian, ())?;
-
-        for entry in &self.entries {
-            entry.write_options(writer, endian, ())?;
-        }
+    // Write empty entries to pad out the rest of the page, except the last
+    // 20 bytes which are zeros instead.
+    #[br(temp)]
+    #[bw(calc = EmptyIndexEntries(
+        Self::total_entries(page_size) - usize::try_from(header.num_entries).unwrap()
```

Comment:
clippy complains about this: "note: converting `u16` to `usize` cannot fail"
```suggestion
        Self::total_entries(page_size) - usize::from(header.num_entries)
```